apps/wallet/src/ui/app/redux/slices/transaction-requests/index.ts

```diff
@@ -221,7 +318,30 @@ const slice = createSlice({
         build.addCase(deserializeTxn.rejected, (state, { payload }) => {
             state.deserializeTxnFailed = true;
         });
+        build.addCase(
+            executeDryRunTransactionRequest.fulfilled,
+            (state, { payload }) => {
+                const { txRequestID, txnMeta, txGasEstimation } = payload;
+
+                if (txnMeta) {
+                    txRequestsAdapter.updateOne(state, {
+                        id: txRequestID,
+                        changes: {
+                            txnMeta,
+                        },
+                    });
+                }
 
+                if (txGasEstimation) {
+                    txRequestsAdapter.updateOne(state, {
```

Comment:
you can build a changes object and do an `updateOne` once

Reply:
got it